Dockerfile

```diff
@@ -1,49 +1,50 @@
 #
-# STEP 1: Prepare environment
+# STEP 1: Prepare environment with cross-compilation tools
 #
-FROM golang:1.22 AS preparer
+FROM --platform=$BUILDPLATFORM golang:1.22 AS preparer
 
-RUN apt-get update                                                        && \
-  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
-  curl \
-  git \
-  zip \
-  unzip \
-  g++ \
-  gcc-aarch64-linux-gnu \
-  bzip2 \
-  make \
-  && rm -rf /var/lib/apt/lists/*
-
-RUN go version
+# Copy our build scripts
+COPY tools/cross-compiler/scripts /tmp/scripts
+RUN chmod +x /tmp/scripts/*.sh
+
+# Install dependencies for cross-compilation
+RUN /tmp/scripts/install_deps.sh
+
+# Create compiler wrappers to handle cross-compilation flags
+RUN /tmp/scripts/create_wrappers.sh
 
 WORKDIR /go/src/github.com/ssvlabs/ssv/
-COPY go.mod .
-COPY go.sum .
+COPY go.mod go.sum ./
```

Comment:
wouldn't `./` include `go.sum` and `go.mod`?

Reply:
optimization pattern - it copies only the dependency files first, runs the mod download, and then copies the rest of the code later. This uses layer caching, so if deps remain the same, we don't need to re download them every time